modules/dmrpp_module/DMZ.cc

```diff
@@ -1544,6 +1557,44 @@ void DMZ::process_chunk(DmrppCommon *dc, const xml_node &chunk) const
     dc->accumlate_storage_size(stoull(size));
 }
 
+void DMZ::process_block(DmrppCommon *dc, const xml_node &chunk,unsigned int block_count) const
+{
+    string href;
+    string trust;
+    string offset;
+    string size;
+    string chunk_position_in_array;
+    string filter_mask;
+    bool href_trusted = false;
+
+    for (xml_attribute attr = chunk.first_attribute(); attr; attr = attr.next_attribute()) {
+        if (is_eq(attr.name(), "href")) {
+            href = attr.value();
+        }
+        else if (is_eq(attr.name(), "trust") || is_eq(attr.name(), "dmrpp:trust")) {
+            href_trusted = is_eq(attr.value(), "true");
+        }
+        else if (is_eq(attr.name(), "offset")) {
+            offset = attr.value();
+        }
+        else if (is_eq(attr.name(), "nBytes")) {
+            size = attr.value();
+        }
+    }
+
+    if (offset.empty() || size.empty())
+        throw BESInternalError("Both size and offset are required for a block node.", __FILE__, __LINE__);
+    if (!href.empty()) {
+        shared_ptr<http::url> data_url(new http::url(href, href_trusted));
```

Comment:
Now that we have C++14 we can use make_shared<http::url>(href, href_trusted) here. The make_shared<>() special form produces a shared_ptr that is quite a bit more efficient than one made using new.